great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -789,28 +785,20 @@ def _add_fluent_datasource(
         if not datasource:
             ds_type = _SourceFactories.type_lookup[kwargs["type"]]
             datasource = ds_type(**kwargs)
-
         assert isinstance(datasource, FluentDatasource)
 
-        datasource._data_context = self
-        datasource._data_context._save_project_config()
-
-        # temporary workaround while we update stores to work better with Fluent Datasources for all contexts
-        # Without this we end up with duplicate entries for datasources in both
-        # "fluent_datasources" and "datasources" config/yaml entries.
-        if self._datasource_store.cloud_mode and not from_config:
-            set_datasource = self._datasource_store.set(key=None, value=datasource)
-            if set_datasource.id:
-                logger.debug(f"Assigning `id` to '{datasource_name}'")
-                datasource.id = set_datasource.id
-
-        self.datasources[datasource_name] = datasource
+        return_obj = self.datasources.set_datasource(
+            name=datasource_name, ds=datasource
+        )
+        assert isinstance(return_obj, FluentDatasource)
+        return_obj._data_context = self
+        self._save_project_config()
 
-        return datasource
+        return return_obj
 
     def _update_fluent_datasource(
         self, datasource: Optional[FluentDatasource] = None, **kwargs
-    ) -> None:
+    ) -> FluentDatasource | BaseDatasource:
```

Comment:
Is this true? Should just be `FluentDatasource` right?

Reply:
Good catch! Addressed in https://github.com/great-expectations/great_expectations/pull/8773/commits/b760677108e3d57b9d8fdc4319398bf8d4f265c4